### PR TITLE
Anim speed factor and refactoring code

### PR DESCRIPTION
**Pull request by AadumKhor**
This commit contains the changes requested in the comments of commit fdf49fd and also changes suggested in #33 (only first point as of now). Updating readme accordingly. Please review @aagarwal1012.

### CODE REVIEW

File: lib/liquid_pull_to_refresh.dart [hunk 1]
```diff
@@ -47,19 +47,18 @@ enum _LiquidPullToRefreshMode {
...
-    this.notificationPredicate = defaultScrollNotificationPredicate,
     this.height,
     this.springAnimationDurationInMilliseconds = 1000,
     this.borderWidth = 2.0,
     this.showChildOpacityTransition = true,
-    this.scrollController,
   })  : assert(child != null),
         assert(onRefresh != null),
-        assert(notificationPredicate != null),
+        assert(animSpeedFactor > 1.0),
```
Comment (aagarwal1012) [inline lib/liquid_pull_to_refresh.dart:61]: @AadumKhor 
Don't you think it should be `animSpeedFactor >= 1.0`?

Comment (AadumKhor) [reply lib/liquid_pull_to_refresh.dart:61]: Yeah. My bad.

File: lib/liquid_pull_to_refresh.dart [hunk 2]
```diff
@@ -356,43 +349,54 @@ class LiquidPullToRefreshState extends State<LiquidPullToRefresh>
...
+                milliseconds: (widget.springAnimationDurationInMilliseconds /
+                        (widget.animSpeedFactor * 5))
+                    .round()),
             curve: Curves.easeOut);
         _showPeakController.value = 0.0;
 
         await _positionController.animateTo(0.0,
             duration: Duration(
-                milliseconds: _kIndicatorSnapDuration.inMilliseconds * 2));
+                milliseconds: (widget.springAnimationDurationInMilliseconds /
```
Comment (aagarwal1012) [inline lib/liquid_pull_to_refresh.dart:397]: @AadumKhor, Did you check that by changing `_kIndicatorSnapDuration` to `springAnimationDurationInMilliseconds`  works fine?

**Comment by aagarwal1012** [comment]
@AadumKhor, please update this branch. Also, Travis CI failed, check via https://travis-ci.com/github/aagarwal1012/Liquid-Pull-To-Refresh/jobs/326418327#L286.

File: lib/liquid_pull_to_refresh.dart (hunk 2 above)
Comment (AadumKhor) [reply lib/liquid_pull_to_refresh.dart:397]: Actually both of them work fine. To keep them all in sync I have changed this value. Looked fine on both emulator and profiler.

**Comment by AadumKhor** [comment]
> 
> 
> @AadumKhor, please update this branch. Also, Travis CI failed, check via https://travis-ci.com/github/aagarwal1012/Liquid-Pull-To-Refresh/jobs/326418327#L286.

It says named parameter 'scrollController' is not defined in main.dart file. There is no named parameter with that name. I have removed it already.

**Comment by aagarwal1012** [comment]
@AadumKhor, I have approved this PR, you can now do a squash merge. After that, if you are done with fixing issues, I will do a version update.

**Comment by AadumKhor** [comment]
@aagarwal1012 it will take some time to fully resolve #33 so I would suggest adding current features in the release while I continue working on other issues.

Also #33 suggests having an option of replacing the custom progress indicator with the built in one. I honestly think otherwise and would like your opinion on it. I am working on the second point though and will come up with a fix. 

Making a squash merge with the current state of code.

**Comment by AadumKhor** [comment]
@aagarwal1012 please update example/lib/main.dart file. It has `scrollController` as a parameter which is incorrect as of last merge.

**Comment by aagarwal1012** [comment]
@AadumKhor, I have updated the example app. 
For the spinning indicator part, we can surely replace it with `CircularProgressIndicator` as it will give better customization. 
Also, I will suggest we can do all the fixes and then we can do a **v2.0.0** release.

**Comment by AadumKhor** [comment]
@aagarwal1012 Sure. I will add the indicator first and then solve the "manually stop refresh" part.

**Comment by AadumKhor** [comment]
@aagarwal1012 The CircularProgressIndicator looks something like this. Right now it is just a boolean to show whether you want to see the built in ProgressIndicator or not. Do we also want to make it completely customizable or is it OK to keep it as a second option that we cannot change? 

It looks something like this right now. 
circular_progress_gif

Looking forward to some feedback.

**Comment by aagarwal1012** [comment]
@AadumKhor, I think we should not change it as there may be some chances which can lead to loss of the essence of the package.  Also, the refresh indicator mentioned in the above image doesn't look good (borders are visible).

**Comment by AadumKhor** [comment]
@aagarwal1012 That is exactly what I thought which is why I asked. And the second point of #33 is also counter-intuitive. If you have initiated the refresh then that process should not stop manually. It is not available in the built in `RefreshIndicator` as well. I guess we should close that issue since those additions are not necessary.

**Comment by aagarwal1012** [comment]
@AadumKhor, I completely agree with you, mention this thing in the issue and close it.

**Pull request merged**